server/app/repository/ApplicationEventRepository.java

```diff
@@ -0,0 +1,54 @@
+package repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+import com.google.common.collect.ImmutableList;
+import io.ebean.DB;
+import io.ebean.Database;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Application;
+import models.ApplicationEvent;
+
+/**
+ * ApplicationEventRepository performs operations on {@link ApplicationEvent} that often involve
+ * other EBean models or asynchronous handling.
+ */
+public class ApplicationEventRepository {
+  private final Database database;
+  private final DatabaseExecutionContext executionContext;
+
+  @Inject
+  public ApplicationEventRepository(DatabaseExecutionContext executionContext) {
+    this.database = DB.getDefault();
+    this.executionContext = checkNotNull(executionContext);
+  }
+
+  /** Insert a new {@link ApplicationEvent} record asynchronously. */
+  public CompletionStage<ApplicationEvent> insert(ApplicationEvent event) {
```

Comment:
At present, the only consumer of this API will be the program admin, where calls can be fully synchronous. Can we convert this to return `ApplicationEvent` and do the toCompletableFuture().join() call in the implementation?

Same goes for getEvents until we add the applicant-facing overload of the method. It's entirely possible that for applicants, we'll just want a `getCurrentStatus` or something along those lines, so we might even end up having a different API entirely exposed for that.

Reply:
done, but removed all the completion stuff and named the method Sync per existing patterns in other Repositories.